source/common/upstream/cluster_manager_impl.h

```diff
@@ -209,11 +210,11 @@ class ClusterManagerImpl : public ClusterManager, Logger::Loggable<Logger::Id::u
 
   struct PrimaryClusterData {
     PrimaryClusterData(uint64_t config_hash, bool added_via_api, ClusterPtr&& cluster)
-        : config_hash_(config_hash), added_via_api_(added_via_api), cluster_(std::move(cluster)) {}
+        : config_hash_(config_hash), added_via_api_(added_via_api), cluster_(cluster.release()) {}
```

Comment:
Fixed, new patches pushed. I had converted an std::unique_ptr to a raw pointer in a factory function which then got leaked when exceptions happened. Reverted that change, all good now :-)